### PR TITLE
External audio device module added to the wrapper

### PR DESCRIPTION
**Pull request by morosev**

### CODE REVIEW

File: windows/wrapper/impl_webrtc_AudioDeviceWasapi.h [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+     AudioDeviceWasapiWeakPtr thisWeak_;
+     mutable zsLib::RecursiveLock lock_;
+
+     IAudioDeviceWasapiDelegateSubscriptions subscriptions_;
+     IAudioDeviceWasapiSubscriptionPtr defaultSubscription_;
+
+     std::string id_;
+
+   private:
+      ScopedCOMInitializer                    _comInit;
```
Comment (robin-raymond) [inline windows/wrapper/impl_webrtc_AudioDeviceWasapi.h:401]: variables should have _ at the end not the start

File: windows/wrapper/impl_webrtc_AudioDeviceWasapi.h [hunk 2]
```diff
@@ -0,0 +1,520 @@
...
+
+      winrt::hstring  _GetDefaultDeviceName(DeviceClass cls);
+      winrt::hstring  _GetListDeviceName(DeviceClass cls, int index);
+      winrt::hstring  _GetDeviceName(DeviceInformation const &device);
+
+      winrt::hstring  _GetListDeviceID(DeviceClass cls, int index);
+      winrt::hstring  _GetDefaultDeviceID(DeviceClass cls);
+      winrt::hstring  _GetDeviceID(DeviceInformation const &device);
+
+      DeviceInformation _GetDefaultDevice(DeviceClass cls, AudioDeviceRole role);
```
Comment (robin-raymond) [inline windows/wrapper/impl_webrtc_AudioDeviceWasapi.h:368]: https://en.cppreference.com/w/cpp/keyword

> Also, all identifiers that contain a double underscore __ in any position and each identifier that begins with an underscore followed by an uppercase letter is always reserved and all identifiers that begin with an underscore are reserved for use as names in the global namespace. See identifiers for more details.

File: windows/wrapper/impl_webrtc_AudioDeviceWasapi.h [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+
+      WAVEFORMATPCMEX*       _mixFormatSurroundOut;
+      bool                   _enableUpmix;
+
+      DeviceInformationCollection  _ptrCaptureCollection;
+      DeviceInformationCollection  _ptrRenderCollection;
+      DeviceInformationCollection  _ptrCollection;
+
+   private:  // WASAPI
+      IAudioClient*                           _ptrClientOut;
```
Comment (robin-raymond) [inline windows/wrapper/impl_webrtc_AudioDeviceWasapi.h:423]: Unless they are initialized in the constructor, these should be defaulted with {} to ensure they become NULL upon creation (true of all basic types). If they are only initialized in the constructor then best to move initialization to the header.

**Review by robin-raymond — CHANGES_REQUESTED** [review]

File: windows/wrapper/impl_webrtc_AudioDeviceWasapi.h (hunk 1 above)
Comment (morosev) [reply windows/wrapper/impl_webrtc_AudioDeviceWasapi.h:401]: Fixed on Sergej/20181206-m71-python

File: windows/wrapper/impl_webrtc_AudioDeviceWasapi.h (hunk 2 above)
Comment (morosev) [reply windows/wrapper/impl_webrtc_AudioDeviceWasapi.h:368]: Fixed on Sergej/20181206-m71-python

File: windows/wrapper/impl_webrtc_AudioDeviceWasapi.h (hunk 3 above)
Comment (morosev) [reply windows/wrapper/impl_webrtc_AudioDeviceWasapi.h:423]: All are initialized in constructor.

**Pull request merged**